### PR TITLE
Update toolkit to CTK 12.6

### PR DESCRIPTION
**Pull request by miscco**
12.6 has been released some time ago, so we should update our containers appropriately

### CODE REVIEW

**Review by bernhardmgruber — APPROVED** [review]

**Comment by bernhardmgruber** [comment]
One question though: My `cuda-12.6/include/crt/host_config.h` indicates that clang 18 should also be supported. Why is it not added by this PR?

File: ci/matrix.yaml [hunk 1]
```diff
@@ -28,22 +28,22 @@ workflows:
...
+    - {jobs: ['build'], project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['gcc13', 'clang16'], cpu: 'arm64'}
+    - {jobs: ['test'],  project: 'cudax', ctk: ['12.0', 'curr'], std: 'all', cxx: ['gcc13']}
     - {jobs: ['test'],  project: 'cudax', ctk: ['12.0'        ], std: 'all', cxx: ['clang14']}
-    - {jobs: ['test'],  project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['clang16']}
+    - {jobs: ['test'],  project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['clang18']}
     # Python jobs:
     - {jobs: ['test'], project: 'pycuda'}
     # cccl-infra:
     - {jobs: ['infra'], project: 'cccl', ctk: '11.1', cxx: ['gcc6', 'clang9']}
-    - {jobs: ['infra'], project: 'cccl', ctk: 'curr', cxx: ['gcc', 'clang']}
```
Comment (bernhardmgruber) [inline ci/matrix.yaml:46]: Suggestion: remove double space

Comment (miscco) [reply ci/matrix.yaml:46]: but that aligns it properly :(

**Review by jrhemstad — COMMENTED** [review]
I assume we also need to update the `devcontainer_version` in the `matrix.yaml`?

**Comment by miscco** [comment]
> I assume we also need to update the `devcontainer_version` in the `matrix.yaml`?

We are still at 24.10

File: ci/matrix.yaml [hunk 2]
```diff
@@ -28,22 +28,22 @@ workflows:
...
+    - {jobs: ['build'], project: 'cudax', ctk: ['12.0'        ], std: 17,    cxx: ['gcc13'], sm: "90"}
+    - {jobs: ['build'], project: 'cudax', ctk: [        'curr'], std: 17,    cxx: ['gcc13'], sm: "90a"}
+    - {jobs: ['build'], project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['gcc13', 'clang16'], cpu: 'arm64'}
+    - {jobs: ['test'],  project: 'cudax', ctk: ['12.0', 'curr'], std: 'all', cxx: ['gcc13']}
     - {jobs: ['test'],  project: 'cudax', ctk: ['12.0'        ], std: 'all', cxx: ['clang14']}
-    - {jobs: ['test'],  project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['clang16']}
+    - {jobs: ['test'],  project: 'cudax', ctk: [        'curr'], std: 'all', cxx: ['clang18']}
     # Python jobs:
-    - {jobs: ['test'], project: 'pycuda'}
+    - {jobs: ['test'], project: 'pycuda', ctk: ['12.5']}
```
[the remark this answers is not in the document]
Comment (miscco) [reply ci/matrix.yaml:43]: Yeah it seems that [pynvjitlink](https://github.com/rapidsai/pynvjitlink) is only built for 12.5 currently

**Review by alliepiper — APPROVED** [review]

**Pull request merged**